instance.tf

```
@@ -0,0 +1,75 @@
+resource "aws_instance" "public_instance" {
```

Do not repeat resource type in resource name (not partially, nor completely):
```suggestion
resource "aws_instance" "public" {
```